src/loam/allocation.rs

```diff
@@ -346,174 +516,164 @@ ascent! {
-
-            prog.input_expr = vec![(WidePtr(Tag::Cons.value(), input),)];
-            prog.run();
-
-            println!("{}", prog.relation_sizes_summary());
-
-            assert_eq!(
-                vec![(WidePtr(Tag::Cons.value(), expected_output),)],
-                prog.output_expr
-            );
-            assert_eq!(cons_count, prog.cons_mem.len());
-            assert_eq!(cons_count, prog.cons_digest_mem.len());
-
-            // This will often fail in the first pass. We need to fix up the memory and provide it as a hint to the
-            // second pass.
-            //assert!(prog.cons_mem_is_contiguous());
-
-            prog
+    fn wide_ptr(tag: LE, digest: [LE; 8]) -> WidePtr {
+        WidePtr(Wide::widen(tag), Wide(digest))
+    }
+
+    fn read_wideptr(zstore: &mut ZStore<BabyBear, LurkChip>, src: &str) -> WidePtr {
+        let ZPtr { tag, digest } = zstore.read(src).unwrap();
+        wide_ptr(tag.elt(), digest)
+    }
+
+    fn test_aux0(
+        zstore: &ZStore<BabyBear, LurkChip>,
+        input: WidePtr,
+        expected_output: WidePtr,
+    ) -> AllocationProgram {
+        let mut prog = AllocationProgram::default();
+
+        prog.import_zstore(zstore);
+        prog.input_expr = vec![(input,)];
+        prog.run();
+
+        println!("{}", prog.relation_sizes_summary());
+
+        assert_eq!(expected_output, prog.output_expr[0].0);
+        prog
+    }
+
+    fn test_aux(input: &str, expected_output: &str) -> AllocationProgram {
+        let mut zstore = lurk_zstore();
+        let input = read_wideptr(&mut zstore, input);
+        let expected_output = read_wideptr(&mut zstore, expected_output);
+        test_aux0(&zstore, input, expected_output)
+    }
+
+    fn test_aux1(input: &str, expected_output: WidePtr) -> AllocationProgram {
+        let mut zstore = lurk_zstore();
+        let input = read_wideptr(&mut zstore, input);
+        test_aux0(&zstore, input, expected_output)
+    }
+
+    fn test_distilled(input: &str, expected_output: &str, bad_input_output: Option<(&str, &str)>) {
+        // Run the first phase
+        let mut zstore = lurk_zstore();
+        let input = read_wideptr(&mut zstore, input);
+        let expected_output = read_wideptr(&mut zstore, expected_output);
+        let original_program = test_aux0(&zstore, input, expected_output);
+
+        // Start the second phase
+        let mut prog = DistilledAllocationProgram::default();
+
+        // Move the hashes (stored in the allocator) and input over
+        prog.allocator = original_program.allocator.clone();
+        prog.input_expr = original_program.input_expr.clone();
+
+        // transfer over the memory (assume it's been distilled)
+        let raw_memory = original_program.export_memory();
+        let mut store = Store::default();
+        let options = DistillationOptions::new().with_summary(0.9);
+        let memory = raw_memory.distill_with_store(&mut store, &options);
```

Comment:
Is the comment above wrong? If this was already distilled, why are we distilling here? If wrong, let's remove the comment on line 624.